x-pack/metricbeat/module/aws/elb/elb_integration_test.go

```diff
@@ -0,0 +1,26 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build integration
+
+package elb
+
+import (
+	"testing"
+
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+	"github.com/elastic/beats/x-pack/metricbeat/module/aws/mtest"
+)
+
+func TestData(t *testing.T) {
+	config, info := mtest.GetConfigForTest("aws", "elb", "300s")
+	if info != "" {
+		t.Skip("Skipping TestData: " + info)
+	}
+
+	metricSet := mbtest.NewReportingMetricSetV2Error(t, config)
+	if err := mbtest.WriteEventsReporterV2Error(metricSet, t, "/"); err != nil {
+		t.Fatal("write", err)
+	}
```

Comment:
This can be implemented now as:
```
metricSet := mbtest.NewFetcher(t, config)
metricSet.WriteEvents(t, "/")
```
So we don't need to modify tests if we change the interface of the metricset (#11998)

Reply:
aha great! thanks for pointing it out!